pkg/controller/clusterdeployment/clusterdeployment_controller_test.go

```diff
@@ -1520,25 +1561,27 @@ func TestMergePullSecrets(t *testing.T) {
 
 	for _, test := range tests {
 		t.Run(test.name, func(t *testing.T) {
-			fakeClient := fake.NewFakeClient(test.existingCD...)
+			if test.globalPullSecret != "" {
+				globalPullSecret := createGlobalPullSecretObj(corev1.SecretTypeDockerConfigJson, globalPullSecret, corev1.DockerConfigJsonKey, test.globalPullSecret)
+				test.existingObjs = append(test.existingObjs, globalPullSecret)
```

Comment:
To test the existence of the global pull secret, the GLOBAL_PULL_SECRET environment variable needs to be set.

Reply:
Right, this what I am missing. Thanks